src/topology/algebra/uniform_ring.lean

```diff
@@ -34,23 +34,20 @@ variables [uniform_add_group α]
 
 lemma continuous_mul : continuous (λ p : completion α × completion α, p.1 * p.2) :=
 begin
-  haveI : is_Z_bilin ((coe ∘ uncurry (*)) : α × α → completion α) :=
-  { add_left := begin
-      introv,
-      change coe ((a + a')*b) = coe (a*b) + coe (a'*b),
-      rw_mod_cast add_mul
-    end,
-    add_right := begin
-      introv,
-      change coe (a*(b + b')) = coe (a*b) + coe (a*b'),
-      rw_mod_cast mul_add
-    end },
-  have : continuous ((coe ∘ uncurry (*)) : α × α → completion α),
-    from (continuous_coe α).comp continuous_mul,
-  convert dense_inducing_coe.extend_Z_bilin _ _ dense_inducing_coe this,
-  { simp only [(*), curry, prod.mk.eta] },
-  { exact completion.is_add_group_hom_coe },
-  { exact completion.is_add_group_hom_coe },
+  let m : α →+ α →+ completion α :=
+  { to_fun := λ a,
+    { to_fun := λ b, ((a*b : α) : completion α),
+      map_zero' := by rw [mul_zero, coe_zero],
+      map_add' := λ x y, by rw_mod_cast [mul_add] },
+  map_zero' := by {ext x, dsimp, rw_mod_cast [zero_mul], refl },
+  map_add' := λ x y, by {ext z, dsimp, rw_mod_cast [add_mul] } },
```

Comment:
I'm a little surprised we don't have somewhere that multiplication in a semiring is bilinear, `mul_bilin : α →+ α →+ α`. If we added that, then this would reduce to `(mul_bilin α).compr₂ to_compl`.

Reply:
(I can only find it for algebras, `algebra.lmul : A →ₐ[R] module.End R A`)

Reply:
I'm being dumb, we have this and I added it in #7382. Untested:

```suggestion
  let m : α →+ α →+ completion α :=
  add_monoid_hom.mul.compr₂ to_compl,
```

Reply:
This leads to type class inference issues.

Reply:
Hmm, you're right. This works:
```suggestion
let m := (add_monoid_hom.mul : α →+ α →+ α).compr₂ to_compl,
```